packages/core/src/loaders/geometryLoader.ts

```diff
@@ -6,31 +6,156 @@ import type {
   IGeometry,
   PublicContourSetData,
   PublicSurfaceData,
+  IGeometryLoadObject,
+  GeometryLoaderFn,
 } from '../types';
 import { createContourSet } from './utils/contourSet/createContourSet';
 import { createSurface } from './utils/surface/createSurface';
+import Events from '../enums/Events';
+import eventTarget from '../eventTarget';
+import triggerEvent from '../utilities/triggerEvent';
 
 interface GeometryOptions {
   type: GeometryType;
   geometryData: PublicContourSetData | PublicSurfaceData;
+  sizeInBytes?: number;
+  segmentIndex?: number;
 }
 
+const geometryLoaders = {};
+let unknownGeometryLoader;
+
 /**
- * Todo: currently we are not targeting loading geometry from a file.
- * This is a placeholder for future work. For instance, separate loaders
- * for .vti, .vtk, .obj, .dat etc. can be created and registered here.
+ * Load a geometry using a registered Cornerstone Geometry Loader.
+ *
+ * The geometry loader that is used will be
+ * determined by the geometry loader scheme matching against the geometryId.
+ *
+ * @param geometryId - A Cornerstone Geometry Object's geometryId
+ * @param options - Options to be passed to the Geometry Loader.
+ *
+ * @returns An Object which can be used to act after a geometry is loaded or loading fails
+ *
  */
+function loadGeometryFromGeometryLoader(
```

Comment:
Consider encapsulating all the functions into a static class, and then export the class - it ties things together better, and allows you to encapsulate things like the geometryLoaders values.  Maybe not worth doing right now, given the timing.